ddns/provider/namesilo.py

```diff
@@ -0,0 +1,162 @@
+
+        if response:
+            # Domain exists, return the domain name as zone_id
+            domain_info = response.get("domain", {})
+            if domain_info:
+                self.logger.debug("Domain found: %s", domain)
+                return domain
+
+        self.logger.warning("Domain not found or not accessible: %s", domain)
+        return None
+
+    def _query_record(self, zone_id, subdomain, main_domain, record_type, line, extra):
+        # type: (str, str, str, str, str | None, dict) -> dict | None
+        """
+        Query existing DNS record
+        @doc: https://www.namesilo.com/api-reference#dns/list-dns-records
+        """
+        response = self._request("dnsListRecords", domain=main_domain)
+
+        if response:
+            records = response.get("resource_record", [])
+            # Handle single record response (API returns dict instead of list)
+            if isinstance(records, dict):
+                records = [records]
+
+            # Find matching record
+            for record in records:
+                if record.get("host") == subdomain and record.get("type") == record_type:
+                    self.logger.debug("Found existing record: %s", record)
+                    return record
+
+        self.logger.debug("No matching record found for %s.%s (%s)", subdomain, main_domain, record_type)
+        return None
+
+    def _create_record(self, zone_id, subdomain, main_domain, value, record_type, ttl, line, extra):
+        # type: (str, str, str, str, str, int | str | None, str | None, dict) -> bool
+        """
+        Create new DNS record
+        @doc: https://www.namesilo.com/api-reference#dns/add-dns-record
+        """
+        response = self._request(
+            "dnsAddRecord", domain=main_domain, rrtype=record_type, rrhost=subdomain, rrvalue=value, rrttl=ttl
+        )
+
+        if response:
+            record_id = response.get("record_id")
+            self.logger.info("DNS record created successfully: %s", record_id)
+            return True
+        else:
+            self.logger.error("Failed to create DNS record")
+            return False
+
+    def _update_record(self, zone_id, old_record, value, record_type, ttl, line, extra):
+        # type: (str, dict, str, str, int | str | None, str | None, dict) -> bool
+        """
+        Update existing DNS record
+        @doc: https://www.namesilo.com/api-reference#dns/update-dns-record
+        """
+        record_id = old_record.get("record_id")
+        if not record_id:
+            self.logger.error("No record_id found in old_record: %s", old_record)
+            return False
+
+        # In NameSilo, zone_id is the main domain name
+        response = self._request(
+            "dnsUpdateRecord",
+            rrid=record_id,
+            domain=zone_id,  # zone_id is main_domain in NameSilo
+            rrhost=old_record.get("host", ""),  # host field contains subdomain
```

Comment:
don't set the default rrhost value let it None if empty in the old_record:
`rrhost=old_record.get("host")`